### DISCUSSION TITLE
Missing 0/0 data in influxDB

### DISCUSSION THREAD

**Discussion by shalak** [alexjustesen/speedtest-tracker#563] [Q&A] [answered]
My ISP was down between 01:06:00 and 03:22:00. The speedtest-tracker reflects this correctly:

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/1536554/234493896-756e2850-6934-43a1-b50d-2f35b6cd7669.png">

However, the values of DL/UL 0/0 is missing from influxDB - there's a gap between 1:00:00 and 4:00:00

![image](https://user-images.githubusercontent.com/1536554/234494396-35390464-66cb-4827-b568-7dc94b8f3fa7.png)

Is this an expected behavior?

**Comment by alexjustesen** [comment]
In my opinion I should probably fix speedtest tracker, as the speed wasn't "0" it's actually a `null` in this case because it didn't complete successfully.

In this case I would save influxdb is right.

**Reply by shalak** [reply]
Hmmm, sure - it's `null`, but one could argue, that from the user perspective they have no internet, thus the speed is 0. But I see your point (might as well there was an issue with speedtest server). I guess [the graphana dashboard](https://grafana.com/grafana/dashboards/17460-speedtest-tracker-v2/) needs to be updated to show missing datapoints as 0, because currently - if you don't look closely at the datapoints - it looks like the connection was UP during that time:

<img width="282" alt="image" src="https://user-images.githubusercontent.com/1536554/234803737-b118a9a9-c2f7-4d90-b124-89a318f23cb5.png">

**Reply by alexdelprete** [reply]
> In my opinion I should probably fix speedtest tracker, as the speed wasn't "0" it's actually a `null` in this case because it didn't complete successfully.

Didn't we discuss this some time ago? I remember we agreed on null, if I'm not mistaken.

**Reply by alexdelprete** [reply]
> it's `null`, but one could argue, that from the user perspective they have no internet, thus the speed is 0.

null is not zero. missing data is not null.

ST should send null, then in influxdb we could use the [fill()](https://docs.influxdata.com/influxdb/cloud/query-data/flux/fill/) function to decide how to plot null values:

![image](https://user-images.githubusercontent.com/7027842/234864952-77b39b5d-3cbb-4a81-8394-502115268249.png)

For missing values, you need interpolation, but that is not available in Flux, see this discussion: https://github.com/influxdata/flux/issues/2428

So if we have nulls, I can adapt the dashboard with `fill(value: 0.0)` otherwise there's not much I can do.

BTW: some users might not agree with null=0 and prefer the current behaviour. We could have an option for this: `send null / send zero / don't send`.

**Reply by alexjustesen** [reply]
> Didn't we discuss this some time ago? I remember we agreed on null, if I'm not mistaken.

Yeah I think we landed on `null` being the right approach.

Ultimately I don't use Graphana or InfluxDB in my stack currently so my knowledge is limited on its functionality

**Reply by alexdelprete** [reply]
I got your back on that, not an expert but a good user. :)

**Comment by alexjustesen** [comment, marked as answer]
Linking this discussion to #549 as a refactored payload for InfluxDB is incoming.
